### PR TITLE
DOC-699: Filtering on paths

### PR DESCRIPTION
**Pull request by diana-macrometa**
This PR breaks Path Filtering and Pruning into separate topics. Content has been heavily edited and rewritten, please edit throughly.

### CODE REVIEW

File: docs/graphs/graph-queries/traversal-queries/filtering-paths.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+```sql
+FOR v, e, p IN 1..5 OUTBOUND 'circles/A' GRAPH 'traversalGraph'
+    FILTER p.edges[*].theTruth ALL == true
+    RETURN { vertices: p.vertices[*]._key, edges: p.edges[*].label }
+```
+
+Or `NONE` of the edges should have `theTruth == true`:
+
+```sql
```
Comment (diana-macrometa) [inline docs/graphs/graph-queries/traversal-queries/filtering-paths.md:60]: @martin-macrometa  - This query returns an empty list. Is that correct? If so, can you provide an example with NONE that does return results?

File: docs/graphs/graph-queries/traversal-queries/filtering-paths.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
+---
+sidebar_position: 25
+title: Filtering Paths
+---
+
+Path filtering enables advanced and efficient filtering capabilities, significantly impacting query performance. Using the `path` variable, you can apply filters based on specific iteration depths. You can filter for absolute positions in the path by specifying a positive number (thus benefiting from optimization strategies), or for relative positions to the path's end by specifying a negative number.
```
Comment (martin-macrometa) [inline docs/graphs/graph-queries/traversal-queries/filtering-paths.md]: impacting -> improving ?

File: docs/graphs/graph-queries/traversal-queries/filtering-paths.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+The following examples are based on the [traversal graph](../../graph-examples/example-graphs#the-traversal-graph). If create this example graph, then you can run any of the queries listed in this topic.
```
Comment (martin-macrometa) [inline docs/graphs/graph-queries/traversal-queries/filtering-paths.md]: "If create this example graph"
missing word
"If you create this example graph"

File: docs/graphs/graph-queries/traversal-queries/filtering-paths.md (hunk 1 above)
Comment (martin-macrometa) [reply docs/graphs/graph-queries/traversal-queries/filtering-paths.md:60]: The result would depend on the data populated in the database. Having an empty result is not a bug.

**Review by martin-macrometa — CHANGES_REQUESTED** [review]

File: docs/graphs/graph-queries/traversal-queries/filtering-paths.md (hunk 1 above)
Comment (diana-macrometa) [reply docs/graphs/graph-queries/traversal-queries/filtering-paths.md:60]: Yeah, it just looks tacky in our documentation. I'll add a note.

File: docs/graphs/graph-queries/traversal-queries/filtering-paths.md (hunk 2 above)
Comment (diana-macrometa) [reply docs/graphs/graph-queries/traversal-queries/filtering-paths.md]: Will it always improve them?

File: docs/graphs/graph-queries/traversal-queries/filtering-paths.md (hunk 1 above)
Comment (diana-macrometa) [reply docs/graphs/graph-queries/traversal-queries/filtering-paths.md:60]: I was hoping there would be an easy fix for the query to make it have results. It's based on the Traversal Graph example.

File: docs/graphs/graph-queries/traversal-queries/filtering-paths.md (hunk 2 above)
Comment (martin-macrometa) [reply docs/graphs/graph-queries/traversal-queries/filtering-paths.md]: Yes, this is the idea.

**Comment by diana-macrometa** [comment]
@martin-macrometa, are you ready to approve this, or do you have more edits?

File: docs/graphs/graph-queries/traversal-queries/filtering-paths.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+## Filtering Edges on the Path
+
+```sql
+FOR v, e, p IN 1..5 OUTBOUND 'circles/A' GRAPH 'traversalGraph'
+    FILTER p.edges[0].theTruth == true
+    RETURN { vertices: p.vertices[*]._key, edges: p.edges[*].label }
+```
+
+This query filters all paths where the starting edge (index 0) has the attribute `theTruth` equal to `true`. The resulting paths will be up to five items long
```
Comment (jamesm-macrometa) [inline docs/graphs/graph-queries/traversal-queries/filtering-paths.md]:
```suggestion
This query filters all paths where the starting edge (index 0) has the attribute `theTruth` equal to `true`. The resulting paths will be up to five items long.
```

File: docs/graphs/graph-queries/traversal-queries/filtering-paths.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    RETURN { vertices: p.vertices[*]._key, edges: p.edges[*].label }
+```
+
+This guarantees that at least one, but possibly more, edges fulfill the condition. You can apply all the above filters to vertices in the same way.
+
+## Comparing Path Filtering vs. Vertex or Edge Filtering
+
+Path filtering affects the iteration on your graph. If certain conditions are not met, then the traversal might stop continuing along this path.
+
+In contrast, filters on vertices or edges only determine whether you're interested in the actual value of these documents. This influences the list of returned documents (if you return `v` or `e`) similar to specifying a non-null `min` value. If you specify a `min` value of 2, then the traversal over the first two nodes of these paths has to be executed - you just won't see them in your result array.
```
Comment (jamesm-macrometa) [inline docs/graphs/graph-queries/traversal-queries/filtering-paths.md]:
```suggestion
In contrast, filters on vertices or edges only determine whether you're interested in the actual value of these documents. This influences the list of returned documents (if you return `v` or `e`) similar to specifying a non-null `min` value. If you specify a `min` value of 2, then you must run the traversal over the first two nodes of these paths - you just won't see them in your result array.
```

File: docs/graphs/graph-queries/traversal-queries/pruning.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
+---
+sidebar_position: 20
+title: Pruning for Efficient Searches
+---
+
+Pruning allows you to create conditions that reduce the amount of data examined during a search, improving query performance and reducing overhead. You can apply pruning to vertices, edges, paths, and any variables defined earlier in the query.
+
+The following examples are based on the [traversal graph](../../graph-examples/example-graphs#the-traversal-graph). If create this example graph, then you can run any of the queries listed in this page.
```
Comment (jamesm-macrometa) [inline docs/graphs/graph-queries/traversal-queries/pruning.md]:
```suggestion
The following examples are based on the [traversal graph](../../graph-examples/example-graphs#the-traversal-graph). If you create this example graph, then you can run any of the queries listed in this page.
```

**Review by jamesm-macrometa — APPROVED** [review]

**Review by martin-macrometa — APPROVED** [review]

**Pull request merged**